mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -883,12 +883,18 @@ def _get_orderby_clauses(order_by_list, session):
             if SearchUtils.is_metric(key_type, "="):
                 clauses.append(
                     sql.case(
-                        [(subquery.c.is_nan.is_(True), 1), (order_value.is_(None), 1)], else_=0
+                        [
+                            (subquery.c.is_nan == sqlalchemy.true(), 1),
```

Comment:
Please add a comment above to indicate something like this:
"Ideally the use of "IS" is preferred here but owing to sqlalchemy translation in MSSQL we are forced to use "=" instead. These 2 options are functionally identical / unchanged because the column (is_nan) is not nullable. However it could become an issue if this precondition changes in the future."